CONTRIBUTING.md

```diff
@@ -38,6 +38,43 @@ For example, for running tests in `multi`, `uuid` and `main` packages, call
 make test-multi test-uuid test-main
 ```
 
+Before running benchmark or measuring performance degradation, install benchmark dependencies:
+```bash
+make bench-deps BENCH_PATH=custom_path
+```
+
+Use the variable `BENCH_PATH` to specialize the path of benchmark artifacts.
+It is set to `bench` by default.
+
+To run benchmark tests, call:
+```bash
+make bench DURATION=5s COUNT=7 BENCH_PATH=custom_path TEST_PATH=.
+```
+
+Use the variable `DURATION` to set the duration of perf tests. That variable is mapped on
+testing [flag](https://pkg.go.dev/cmd/go#hdr-Testing_flags) `-benchtime` for gotest.
+It may take the values in seconds (e.g, `5s`) or count of iterations (e.g, `1000x`).
+It is set to `3s` by default.
+
+Use the variable `COUNT` to control the count of benchmark runs for each test. It is set to `5` by default.
+That variable is mapped on testing flag `-count`.
+Use higher values if the benchmark numbers aren't stable.
+
+Use the variable `TEST_PATH` to set the directory of test files.
+It is set to `./...` by default, so it runs all the Benchmark tests in project.
+
+Before calculating performance difference between master and the current branch, 
+prepare benchmark results of master:
+```bash
+make perf-reference DURATION=5s COUNT=7 BENCH_PATH=custom_path TEST_PATH=.
+```
+
+To measure performance degradation after changes in code, firstly call `make bench`
+and `perf-reference` and then run:
```

Comment:
```suggestion
and `make perf-reference` and then run:
```

Or maybe write a single command `make bench perf-reference perf-diff`

Reply:
Yes, that will work. Sounds better, fixed.